Libraries/LibGUI/GTextEditor.h

```diff
@@ -172,6 +182,91 @@ class GTextEditor
     Rect visible_text_rect_in_inner_coordinates() const;
     void recompute_all_visual_lines();
 
+    class UndoCommand {
+        //friend class GTextEditor;
+
+    public:
+        UndoCommand(GTextEditor& text_editor);
+        // GTextEditor& text_editor() const;
+        virtual ~UndoCommand();
+        virtual void apply();
```

Comment:
Seems like `undo()` and `redo()` would be better names than `apply()` and `unapply()` actually.
Since `apply()` is actually *un*doing something.